webpack/worker/webpack.config.js

```diff
@@ -0,0 +1,113 @@
+const {join} = require('path');
+
+function notAvailable() {
+  throw new Error(`Watcher polling or pausing is not available under bazel worker mode.`);
+}
+
+class WorkerWatchFileSystem {
+  constructor(inputFileSystem) {
+    this.inputFileSystem = inputFileSystem;
+  }
+
+  watch(files, directories, missing, startTime, options, callback) {
+    // We do not care about what webpack tells us to watch.
+    // the source of truth is bazel and bazel tells us which files
+    // are present in the build and when they are chaned or removed.
+    if (!files || typeof files[Symbol.iterator] !== 'function') {
+      throw new Error('Invalid arguments: \'files\'');
+    }
+    if (!directories || typeof directories[Symbol.iterator] !== 'function') {
+      throw new Error('Invalid arguments: \'directories\'');
+    }
+    // Missing files are files which are needed to run the build but were not found
+    // in the filesystem. So webpack reports these files via separate set so that the
+    // watcher knows that this file does not exist and does something else other than
+    // opening a file watcher against the missing files.
+    if (!missing || typeof missing[Symbol.iterator] !== 'function') {
+      throw new Error('Invalid arguments: \'missing\'');
+    }
+    if (typeof callback !== 'function') {
+      throw new Error('Invalid arguments: \'callback\'');
+    }
+    if (typeof options !== 'object') {
+      throw new Error('Invalid arguments: \'options\'');
+    }
+
+    if (options.poll) {
+      notAvailable();
+    }
+
+    const rootPath = process.cwd();
+    /** @type {Map<string, { safeTime: number, timestamp: number }} */
+    const mtimeMap = new Map();
+    /** @type {Map<string, string} */
+    const digestMap = new Map();
+
+    /** @param inputs {{[input: string]: string}} */
```

Comment:
Just a general comment for this file. Should we rewrite it in typescript? I say this only because you've added type annotations throughout but they aren't be run through a compiler

Reply:
this was my suggestion, authoring in typescript will make this repo a bit more complex to build and publish

we could still typecheck with tsc --allowJs maybe

Reply:
I agree. right now the whole publishing process is one command: `npm pack`